src/resultsdisassemblypage.cpp

```diff
@@ -171,6 +172,37 @@ ResultsDisassemblyPage::ResultsDisassemblyPage(CostContextMenu* costContextMenu,
         showDisassembly();
     });
 
+    auto searchShortcut = new QShortcut(QKeySequence::Find, this);
+    connect(searchShortcut, &QShortcut::activated, this, [this] {
+        ui->searchWidget->setVisible(true);
+        ui->searchEdit->setFocus();
+    });
+
+    connect(ui->closeSearch, &QPushButton::pressed, this, [this] { ui->searchWidget->setVisible(false); });
+
+    connect(ui->nextSearchResult, &QPushButton::pressed, this,
+            [this] { m_sourceCodeModel->findForward(ui->searchEdit->text(), m_searchResultIndex); });
+
+    connect(ui->prevSearchResult, &QPushButton::pressed, this,
+            [this] { m_sourceCodeModel->findBackwards(ui->searchEdit->text(), m_searchResultIndex); });
+
+    connect(m_sourceCodeModel, &SourceCodeModel::searchResultFound, this, [this](const QModelIndex& index) {
+        m_searchResultIndex = index;
+        ui->sourceCodeView->scrollTo(m_searchResultIndex, QAbstractItemView::ScrollHint::PositionAtCenter);
+        ui->searchResultWarning->hide();
+    });
+
+    // search if enter is pressed
+    connect(ui->searchEdit, &QLineEdit::returnPressed, ui->nextSearchResult, &QPushButton::pressed);
+
+    // I don't know how to show an error so I use a warning widget
+    // kate, ... set the backgroundcolor to red but I don't know how they do this
+    // google say ->setStylesheet("background: red");
```

Comment:
no, stylesheets are bad as our KDAB blog should have thought you! we can set a palette with a negative foreground text color, KColorScheme even has API for that afair